### PR TITLE
Springboot MVC and Springboot REST API projects

### PR DESCRIPTION
**Pull request by stamptime**

### CODE REVIEW

File: springboot-rest-api/src/main/java/org/devops/bootcamp/repositories/impl/OrderRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @Override
+    @Transactional
+    public Order delete(Order entity) {
+        this.entityManager.remove(entityManager.contains(entity) ? entity : entityManager.merge(entity));
+        return entity;
+    }
+
+    @Override
+    public List<Order> getAll() {
+        return entityManager.createQuery("SELECT o FROM Order o", Order.class).getResultList();
```
Comment (roenfa) [inline springboot-rest-api/src/main/java/org/devops/bootcamp/repositories/impl/OrderRepositoryImpl.java:38]: I would prefer that you use a repository to get the data from DB and not use the queries in each method, I mean embed the SQL queries into the code is not a good practice

File: springboot-rest-api/src/test/java/org/devops/bootcamp/controllers/AuthenticationControllerTests.java [hunk 2]
```diff
@@ -1,41 +1,75 @@
...
 
     @Autowired
-    private AuthenticationManager authenticationManager;
+    public AuthenticationControllerTests(MockMvc mockMvc, AuthenticationController authenticationController) {
+        this.mockMvc = mockMvc;
+        this.authenticationController = authenticationController;
+    }
 
     @Test
     public void existentUserCanGetTokenAndAuthentication() throws Exception {
```
Comment (roenfa) [inline springboot-rest-api/src/test/java/org/devops/bootcamp/controllers/AuthenticationControllerTests.java:37]: Does it run without problems?

**Comment by roenfa** [comment]
@VallecillaJesus Good job!
MVC project -> you forgot the implementation of contacts jsp files (update/delete)
REST API project -> I was not able to see the connection to the DB, I would prefer to use a repository (JpaRepository for example, as we did with h2)
And also, you need to complete the unit tests for services and controllers at least

File: springboot-rest-api/src/main/java/org/devops/bootcamp/repositories/impl/OrderRepositoryImpl.java (hunk 1 above)
Comment (stamptime) [reply springboot-rest-api/src/main/java/org/devops/bootcamp/repositories/impl/OrderRepositoryImpl.java:38]: Ok, I will take it into account, I thought it was a good idea in order to follow Open/Close SOLID principle, but now i see that it wasn't

File: springboot-rest-api/src/test/java/org/devops/bootcamp/controllers/AuthenticationControllerTests.java (hunk 2 above)
Comment (stamptime) [reply springboot-rest-api/src/test/java/org/devops/bootcamp/controllers/AuthenticationControllerTests.java:37]: Sure, it runs properly.

**Comment by stamptime** [comment]
Thanks for the feedback, i'll take them all into account and it's true, there are some features that I didn't achieve as you mentioned.

**Pull request closed without merging**